sc/src/main/java/uk/ac/ebi/atlas/experimentimport/ScxaExperimentDao.java

```diff
@@ -104,4 +106,12 @@ private ExperimentDTO getSingleExperiment(List<ExperimentDTO> experimentDtos, St
         checkExperimentFound(experimentDtos.size() == 1, accession);
         return experimentDtos.get(0);
     }
+
+    // Returns comma-separated IDs, or null if there are no IDs
+    private String getPublicationIdsAsString(Collection<String> publicationIds) {
+        return publicationIds.stream()
+                .collect(Collectors.collectingAndThen(
+                        Collectors.joining(", "),
+                        (result) -> result.isEmpty() ? null : result));
```

Comment:
Why do you need to return `null` if the resulting string is empty? Have a look at `ScxaExperimentDtoResultSetExtractor` because the logic is now scattered across these two classes.

Reply:
So I would argue that from now on we should add NULL to the database in case we don't have PubMed IDs and I'll change the `ResultSetExtractor` class to check for null or empty string (as existing experiments have empty strings).

Reply:
I agree that NULL is more correct than an empty string. Sticking to one convention, and a better one, is definitely the best choice.